pkg/networkservice/core/trace/server_bench_test.go

```diff
@@ -0,0 +1,173 @@
+	res := testutil.NewConnection()
+
+	res.GetConnection().Path = &networkservice.Path{}
+	res.GetConnection().Labels = make(map[string]string)
+
+	for i := 0; i < size; i++ {
+		res.GetConnection().GetPath().PathSegments = append(res.GetConnection().GetPath().PathSegments,
+			&networkservice.PathSegment{
+				Name:  nanoid.MustGenerateString(15),
+				Token: nanoid.MustGenerateString(25),
+				Id:    nanoid.MustGenerateString(20),
+			})
+	}
+
+	res.GetConnection().Id = res.GetConnection().GetPath().GetPathSegments()[0].Id
+
+	return res
+}
+
+func newTestServerChain(size int) networkservice.NetworkServiceServer {
+	var servers []networkservice.NetworkServiceServer
+
+	for i := 0; i < size; i++ {
+		id := i
+		servers = append(servers, checkrequest.NewServer(nil, func(_ *testing.T, nsr *networkservice.NetworkServiceRequest) {
+			nsr.GetConnection().GetLabels()["common-label"] = fmt.Sprint(id)
+		}))
+	}
+
+	return chain.NewNetworkServiceServer(servers...)
+}
+
+func newTestStaticServerChain(size int) networkservice.NetworkServiceServer {
+	var servers []networkservice.NetworkServiceServer
+
+	for i := 0; i < size; i++ {
+		servers = append(servers, null.NewServer())
+	}
+
+	return chain.NewNetworkServiceServer(servers...)
+}
+
+func Benchmark_ShortRequest_Info(b *testing.B) {
+	var s = newTestServerChain(10)
+	var request = newTestRequest(5)
+	var ctx = context.Background()
+
+	logrus.SetLevel(logrus.InfoLevel)
+	logrus.SetOutput(io.Discard)
+	log.EnableTracing(true)
+
+	b.ResetTimer()
+
+	for range b.N {
+		_, _ = s.Request(ctx, request)
+	}
+}
+
+func Benchmark_LongRequest_Info(b *testing.B) {
+	var s = newTestServerChain(100)
+	var request = newTestRequest(100)
+	var ctx = context.Background()
+
+	logrus.SetLevel(logrus.InfoLevel)
+	logrus.SetOutput(io.Discard)
+	log.EnableTracing(true)
+
+	b.ResetTimer()
+
+	for range b.N {
+		_, _ = s.Request(ctx, request)
+	}
+}
+
+func Benchmark_ShortRequest_Trace(b *testing.B) {
+	var s = newTestServerChain(10)
+	var request = newTestRequest(5)
+	var ctx = context.Background()
+
+	logrus.SetLevel(logrus.TraceLevel)
+	logrus.SetOutput(io.Discard)
+	log.EnableTracing(true)
+
+	b.ResetTimer()
+
+	for range b.N {
+		_, _ = s.Request(ctx, request)
+	}
+}
+func Benchmark_LongRequest_Trace(b *testing.B) {
+	var s = newTestServerChain(100)
+	var request = newTestRequest(100)
+	var ctx = context.Background()
+
+	logrus.SetLevel(logrus.TraceLevel)
+	logrus.SetOutput(io.Discard)
+	log.EnableTracing(true)
+
+	b.ResetTimer()
+
+	for range b.N {
+		_, _ = s.Request(ctx, request)
+	}
+}
+
+func Benchmark_LongRequest_Trace_NoDiff(b *testing.B) {
+	var s = newTestStaticServerChain(100)
+	var request = newTestRequest(100)
+	var ctx = context.Background()
+
+	logrus.SetLevel(logrus.TraceLevel)
+	logrus.SetOutput(io.Discard)
+	log.EnableTracing(true)
+
+	b.ResetTimer()
+
+	for range b.N {
+		_, _ = s.Request(ctx, request)
+	}
+}
+
+func Benchmark_LongRequest_Diff_Warn(b *testing.B) {
```

Comment:
Is it a proper name for this test? Shouldn't it be `Benchmark_LongRequest_NoDiff_Warn`?

Reply:
Thanks, fixed.